hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/AddOMSubcommand.java

```diff
@@ -0,0 +1,174 @@
+ *  limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.admin.om;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import org.apache.hadoop.hdds.cli.HddsVersionProvider;
+import org.apache.hadoop.hdds.conf.OzoneConfiguration;
+import org.apache.hadoop.hdds.protocol.proto.HddsProtos;
+import org.apache.hadoop.net.NetUtils;
+import org.apache.hadoop.ozone.OmUtils;
+import org.apache.hadoop.ozone.client.OzoneClientException;
+import org.apache.hadoop.ozone.om.helpers.ServiceInfo;
+import org.apache.hadoop.ozone.om.protocol.OzoneManagerProtocol;
+import org.apache.hadoop.ozone.protocol.proto.OzoneManagerProtocolProtos.OMNodeInfo;
+import org.apache.hadoop.ozone.protocol.proto.OzoneManagerProtocolProtos.OMRoleInfo;
+import org.apache.hadoop.util.StringUtils;
+import picocli.CommandLine;
+
+import java.util.concurrent.Callable;
+
+import static org.apache.hadoop.ozone.om.OMConfigKeys.OZONE_OM_ADDRESS_KEY;
+import static org.apache.hadoop.ozone.om.OMConfigKeys.OZONE_OM_HTTPS_ADDRESS_KEY;
+import static org.apache.hadoop.ozone.om.OMConfigKeys.OZONE_OM_HTTP_ADDRESS_KEY;
+import static org.apache.hadoop.ozone.om.OMConfigKeys.OZONE_OM_NODES_KEY;
+import static org.apache.hadoop.ozone.om.OMConfigKeys.OZONE_OM_RATIS_PORT_DEFAULT;
+import static org.apache.hadoop.ozone.om.OMConfigKeys.OZONE_OM_RATIS_PORT_KEY;
+
+/**
+ * Handler of om roles command.
+ */
+@CommandLine.Command(
+    name = "addom", aliases = "addoms, addnewom, addnewoms",
+    description = "Adds newly Bootstrapped OM(s) to the cluster. " +
+        "\nNote - New OM(s) should be started in bootstrap mode before " +
+        "running this command." +
+        "\nNote - Configuration files of all existing " +
+        "OM(s) MUST be updated with information about the new OM(s) (" +
+        OZONE_OM_NODES_KEY + ", " + OZONE_OM_ADDRESS_KEY + ", " +
+        OZONE_OM_HTTP_ADDRESS_KEY + ", " + OZONE_OM_HTTPS_ADDRESS_KEY + ", " +
+        OZONE_OM_RATIS_PORT_KEY + ").",
+    mixinStandardHelpOptions = true,
+    versionProvider = HddsVersionProvider.class)
+public class AddOMSubcommand implements Callable<Void> {
+
+  @CommandLine.ParentCommand
+  private OMAdmin parent;
+
+  @CommandLine.Option(names = {"-id", "--service-id"},
+      description = "OM Service ID",
+      required = true)
+  private String omServiceId;
+
+  @CommandLine.Option(names = {"-nodeid", "-nodeids", "--new-node-ids"},
+      description = "A comma separated list of node IDs of the new OM(s). ",
+      required = true)
+  private String newNodeIds;
+
+  private OzoneManagerProtocol ozoneManagerClient;
+
+  @Override
+  public Void call() throws Exception {
+    boolean success = false;
+    try {
+      ozoneManagerClient =  parent.createOmClient(omServiceId);
+
+      List<OMNodeInfo> newOMNodeInfos = getNewNodeInfos();
+
+      if (!newOMNodeInfos.isEmpty()) {
```

Comment:
Here first we need to check for not null, as in few code paths getNewNodeInfos returns null.

Reply:
Done.